### PR TITLE
Issue #630: releasenotes-builder: update Github template to be more c…

### LINKED ISSUE #630: releasenotes-builder: update Github template to be more compact
now:
```
Checkstyle 10.4 - https://checkstyle.org/releasenotes.html#Release_10.4



Bug fixes:

  #11736 - MissingJavadocType: Support qualified annotation names
  #11655 - Update google_checks.xml to have the SuppressionCommentFilter ...

<details>
<summary>Other Changes:</summary>
<br>
  Update releasenotes to use ...
<br />
  Allow SuppressWarningHolder ....
```

expected ( attention to br tags):
```
Checkstyle 10.4 - https://checkstyle.org/releasenotes.html#Release_10.4

Bug fixes:

  #11736 - MissingJavadocType: Support qualified annotation names
  #11655 - Update google_checks.xml to have the SuppressionCommentFilter ....

<details>
<summary>Other Changes:</summary>
<br/>
  Update releasenotes to ... <br/>
  Allow SuppressWarningHolder ... <br/>

```

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Fixes #630

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
I am not sure if I have to give the path location of github_post.template or generated template of github:
https://github.com/checkstyle/contribution/pull/655/files#diff-f0cfbc41ea611a4b309a6bb172b63275aef9606a0c0d36a37b93f4f655e61e38R234

**Review by romani — CHANGES_REQUESTED** [review]
There should be functional changes, not only tests

**Comment by Rahulkhinchi03** [comment]
@romani What kind of functional changes?

**Comment by romani** [comment]
Different location of BR tags, see issue description https://github.com/checkstyle/contribution/issues/630

**Comment by Rahulkhinchi03** [comment]
@romani But that is already resolved by https://github.com/checkstyle/contribution/pull/632

**Comment by romani** [comment]
There is extra BR tag
```
<details>
<summary>Other Changes:</summary>
<br/>
  Kill surviving mutations in pitest-api <br/>
<br/>
  Solve fb-contrib errors <br/>
<br/>
  update code
```

**Comment by Rahulkhinchi03** [comment]
Screenshot from 2022-09-11 20-39-20-1

**Comment by Rahulkhinchi03** [comment]
As there is a lot of space in between of release and notes, the test will always fail as there are empty lines

File: releasenotes-builder/src/main/resources/com/github/checkstyle/templates/github_post.template [hunk 1]
```diff
@@ -28,7 +28,6 @@ Bug fixes:
 <details>
 <summary>Other Changes:</summary>
 <#list notesMessages as message>
-<br/>
```
Comment (romani) [inline releasenotes-builder/src/main/resources/com/github/checkstyle/templates/github_post.template:31]: After summary tag, we need BR tag.
Please attentively read issue description.

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Comment by Rahulkhinchi03** [comment]
Screenshot from 2022-09-11 21-21-41

File: releasenotes-builder/src/main/resources/com/github/checkstyle/templates/github_post.template (hunk 1 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/main/resources/com/github/checkstyle/templates/github_post.template:31]: Changes Done

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 2]
```diff
@@ -229,6 +229,15 @@ public void testGenerateOnlyGitHub() throws Exception {
         assertFile("githubPageBreakingCompatibility.txt", MainProcess.GITHUB_FILENAME);
     }
 
+    @Test
+    public void testNoEmptyLineInGitHub() throws Exception {
+        final String lines = getFileContents(new File(MainProcess.GITHUB_FILENAME));
```
Comment (romani) [inline releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:234]: you need to run `MainProcess.runPostGenerationAndPublication`
and validate something like
`assertFile("githubPageBreakingCompatibility.txt", MainProcess.GITHUB_FILENAME);`

to let me see generated content in gituhb as file.

**Review by romani — CHANGES_REQUESTED** [review]
item:

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 2 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:234]: but it says that no such file exists. So should I use the `full path` or `MainProcess.GITHUB_FILENAME`

Comment (romani) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Use full path, I just gave example of how file content is validated

Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Used, please check  the pushed code

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 3]
```diff
@@ -229,6 +229,27 @@ public void testGenerateOnlyGitHub() throws Exception {
...
+        Assert.assertEquals("no errors", 0, errors.size());
+
+        final String lines = getFileContents(new File("src/main/resources/com/github/checkstyle/"
+           + "templates/github_post.template"));
+        final String[] linesList = lines.split("\\r?\\n");
+        for (String line : linesList) {
+            Assert.assertFalse("line should not be empty", line.isEmpty());
+        }
+        assertFile(MainProcess.XDOC_FILENAME);
+        assertFile(MainProcess.TWITTER_FILENAME);
```
Comment (romani) [inline releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:247]: Remove unrelated files.

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 4]
```diff
@@ -229,6 +229,27 @@ public void testGenerateOnlyGitHub() throws Exception {
...
+        assertFile(MainProcess.RSS_FILENAME);
+        assertFile(MainProcess.MLIST_FILENAME);
+        assertFile("githubPageBreakingCompatibility.txt", MainProcess.GITHUB_FILENAME);
```
Comment (romani) [inline releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:250]: Put here file with content that you expect

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 3 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Changes Done

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 4 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Changes Done

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 5]
```diff
@@ -229,6 +229,23 @@ public void testGenerateOnlyGitHub() throws Exception {
...
+            createNotes(createAllNotes(), null, null, null, null), createBaseCliOptions()
+                .setGenerateGitHub(true).build());
+
+        Assert.assertEquals("no errors", 0, errors.size());
+
+        final String lines = getFileContents(new File("src/main/resources/com/github/checkstyle/"
+           + "templates/github_post.template"));
+        final String[] linesList = lines.split("\\r?\\n");
+        for (String line : linesList) {
+            Assert.assertFalse("line should not be empty", line.isEmpty());
```
Comment (romani) [inline releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:244]: Please remove such strange validation on empty lines, it is not required.
All that you need is validation of githubPageAll.txt

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 5 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: then how we will be able to test the empty lines? As run `runPostGenerationAndPublication` runs on breaking compatibility text file

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 6]
```diff
@@ -229,6 +229,16 @@ public void testGenerateOnlyGitHub() throws Exception {
         assertFile("githubPageBreakingCompatibility.txt", MainProcess.GITHUB_FILENAME);
     }
 
+    @Test
+    public void testNoEmptyLineInGitHub() throws Exception {
```
Comment (romani) [inline releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java:233]: Fix name to match what it actually do

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java (hunk 6 above)
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Changes Done

File: releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java [hunk 7]
```diff
@@ -229,6 +229,16 @@ public void testGenerateOnlyGitHub() throws Exception {
         assertFile("githubPageBreakingCompatibility.txt", MainProcess.GITHUB_FILENAME);
     }
 
+    @Test
+    public void testEmptyLineInGitHub() throws Exception {
```
[the remark this answers is not in the document]
Comment (Rahulkhinchi03) [reply releasenotes-builder/src/test/java/com/github/checkstyle/templates/TemplateProcessorTest.java]: Changes Done

**Review by romani — CHANGES_REQUESTED** [review]
Please make CI green https://ci.appveyor.com/project/Checkstyle/contribution/builds/44751986/job/ad7uyjvrhn1o60fk#L91

**Comment by Rahulkhinchi03** [comment]
@romani githubPageAll and github_post.template will never be same, I think we have to go with breaking compatibility file only

**Review by romani — APPROVED** [review]
ok to merge
